src/reducers/config.js

```diff
@@ -1,14 +1,21 @@
-import Immutable from 'immutable';
```

Comment:
Causes for change in this file:

* `action.payload` is an immutable map instead of an object.
* isFetching is set to `true` by default.
* State can be merged prior to configuration finalization.